### PR TITLE
Use SLEEF functions for erf/exp on macOS ARM64

### PR DESCRIPTION
**Pull request by danieldk**
The NEON `Vectorized<float>` implementation does not use SLEEF functions in order to compile on mobile platforms. However, SLEEF is already compiled on macOS ARM64 and is safe to use on that platform.

Since transformers rely heavily on `erf` (GELU) and `exp` (softmax), using the SLEEF implementations of `erf`/`exp` can provide large speedups of transformers on Apple M1 Macs. For instance, we have found spaCy transformer models to be 20% faster when using the SLEEF implementation of these functions.

This PR is a follow-up to #70354, but with a more limited scope as suggested in https://github.com/pytorch/pytorch/pull/70354#pullrequestreview-852154370.

Profile of a tranformer model before this change:

Screen Shot 2022-01-14 at 09 37 56

Most time is spent in `erff`/`expf`.

Profile of the same transformer model after this change:

Screen Shot 2022-01-14 at 09 39 39

Time spent in `erff` is reduced from 1.94min to 1.18min, time spent in `expf` is reduced from 60s to 8s.

### CODE REVIEW

**Review by malfet — APPROVED** [review]

**Comment by danieldk** [comment]
Superseded by merging of #70354.
